include/revng/Support/TemporaryFile.h

```diff
@@ -14,20 +14,47 @@ class TemporaryFile {
 private:
   llvm::SmallString<32> Path;
 
+  TemporaryFile(const llvm::StringRef Path, int /* dummy */) :
+    Path(Path.str()) {
+    llvm::sys::RemoveFileOnSignal(Path);
+  }
```

Comment:
I see some non-trivial changes to `TemporaryFile` in a commit that's not strictly related.
What's going on here?

Reply:
1. Added `TemporaryFile::make` for the cases where you want to catch the `error_code` if it happens (with the weird dummy constructor to avoid ambiguity)
2. Fixed the move assignment, since it would leak the original file if it was present
3. Fixed the destructor, otherwise it would be called even on a moved object

Reply:
Gotcha, a commit would be good.

Reply:
Split off to a separate commit